source/material_model/steinberger.cc

```diff
@@ -489,6 +529,53 @@ namespace aspect
               seismic_out->vs[i] = seismic_Vs(in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
             }
         }
+
+      /* We separate the calculation of specific heat and thermal expansivity,
+        * because they may depend on cell-wise averaged values that are only
+        * available here.
+        */
```

Comment:
You only need any of the things you do below if latent_heat = true, correct? Why not put the `if (latent_heat)` statement around everything or return if latent_heat = false?

Reply:
Oh, and could you add the equations how alpha and cp are computed based on H? Just in the comment above, like `c_p = dH/dT`or something like that...